pkg/cloud/services/provider/provider.go

```diff
@@ -85,7 +85,7 @@ func newClient(cloud clientconfig.Cloud, caCert []byte) (*gophercloud.ProviderCl
 
 	opts, err := clientconfig.AuthOptions(clientOpts)
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, fmt.Errorf("auth option failed: %v for cloud: %v", err, cloud)
```

Comment:
afaik it's best practise to append the error at the end
```suggestion
		return nil, nil, fmt.Errorf("auth option failed for cloud %s: %v", cloud, err)
```

Reply:
I would also suggest just printing the cloud name as I think we would be leaking credentials in the log otherwise

Reply:
done :)